lua/telescope/builtin/files.lua

```diff
@@ -145,106 +145,126 @@ files.grep_string = function(opts)
   }):find()
 end
 
--- TODO: Maybe just change this to `find`.
--- TODO: Support `find` and maybe let people do other stuff with it as well.
-files.find_files = function(opts)
+local function which_find_command(opts)
+  local command = opts.find_command
+  if command then
+    return command[1]
+  end
+  if 1 == vim.fn.executable "fd" then
+    return "fd"
+  elseif 1 == vim.fn.executable "fdfind" then
+    return "fdfind"
+  elseif 1 == vim.fn.executable "rg" then
+    return "rg"
+  elseif 1 == vim.fn.executable "find" and vim.fn.has "win32" == 0 then
+    return "find"
+  elseif 1 == vim.fn.executable "where" then
+    return "where"
+  end
+end
+
+local function find_command_options(opts)
+  local command = which_find_command(opts)
   local find_command = opts.find_command
   local hidden = opts.hidden
   local no_ignore = opts.no_ignore
   local follow = opts.follow
   local search_dirs = opts.search_dirs
 
-  if search_dirs then
-    for k, v in pairs(search_dirs) do
-      search_dirs[k] = vim.fn.expand(v)
+  if command == "fd" then
```

Comment:
Cant we refactor this now ? if `command == "fd" or command == "fd_find" or command == 'rg'` all do the same if i am not wrong

Reply:
`fd` and `fdfind` apparently are the same, `rg` is diferent.

Reply:
But I could not say they will be the same in every option they have. I only know `fd`, I wont put my hands in the fire for the other commads, that's why I didn't want to change anything there.

Reply:
`fd` and `fdfind` are 100% the same tool. the command is just called `fdfind` on ubuntu vs `fd` on all other systems (arch, nix, ...)

Reply:
Yeah i see that rg is slightly different. We could still pull it in like this

```lua
if command == "fd" or command == "fd_find" or command == "rg" then
  if hidden then
    table.insert(find_command, "--hidden")
  end
  if no_ignore then
    table.insert(find_command, "--no-ignore")
  end
  if follow then
    table.insert(find_command, "-L")
  end
  if search_dirs then
    if command ~= "rg" then
      table.insert(find_command, ".")
    end
    for _, v in pairs(search_dirs) do
      table.insert(find_command, v)
    end
  end
else if ...
```
cant we?

Reply:
Haha, I just saw that [here](https://github.com/sharkdp/fd#on-ubuntu), let me fix that.